obs-studio-server/source/nodeobs_service.cpp

```diff
@@ -626,8 +626,15 @@ int OBS_service::resetVideoContext(bool reload)
 		if (ovi.base_width > 1280 && ovi.base_height > 720) {
 			int idx = 0;
 			do {
-				ovi.output_width  = uint32_t(double(ovi.base_width) / vals[idx]);
-				ovi.output_height = uint32_t(double(ovi.base_height) / vals[idx]);
+				double use_val = 1.0;
+				if(idx < numVals)
+				{
+					use_val = vals[idx];
+				} else {
+					use_val = 3.0 + double(numVals - idx + 1)/2.0;
```

Comment:
I would use `vals[numVals-1]` instead `3.0` just in case we decide to change the values someday.

Reply:
agreed.